### PR TITLE
#159987410 Users should be able to get notifications

### PR DESCRIPTION
**Pull request by emmaadesile**
#### What does this PR do?
Implement notifications

#### Description of Task to be completed?
- modify the database relationship
- implement the functionality to create notifications
- implement the functionality to push notifications
- implement the functionality to view notifications and mark as read

#### How should this be manually tested?
- Clone the repository.
- Check out the `ft-notifications-159987410`  branch.
- Run `npm install` on your local machine.
- Run the db migrations using `npm run migrate`.
- Start the server using `npm run start_dev`.
- Using postman or any similar. get  a token upon registration and use it to test these routes:
**GET** `localhost:8000/api/v1/users/notifications` to get all notifications
**GET** `localhost:8000/api/v1/users/notifications/:notifyId` to get a single notification
**PUT** `localhost:8000/api/v1/users/notifications` to mark all as read
**PUT** `localhost:8000/api/v1/users/notifications/:notifyId` to mark one as read

#### Any background context you want to provide?
This feature will enable a user to get, view and mark notifications as read

#### What are the relevant pivotal tracker stories?
 [159987410](https://www.pivotaltracker.com/story/show/159987410)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,9 +6,9 @@
   "scripts": {
     "pretest": "NODE_ENV=test sequelize db:migrate && NODE_ENV=test sequelize db:seed:all && NODE_ENV=test npm run create_admin",
     "test": "NODE_ENV=test nyc mocha test/**.spec.js --require babel-polyfill --require babel-core/register --exit",
-    "posttest": "NODE_ENV=test sequelize db:migrate:undo:all",
+    "posttests": "NODE_ENV=test sequelize db:migrate:undo:all",
```
Comment (johngorithm) [inline package.json]: I would like to know why you altered the above script's name

File: server/controllers/notificationController.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        }
+      })
+      .then((notifications) => {
+        const count = notifications.length;
+        return res.status(200).jsend.success({
+          count,
+          notifications
+        });
+      })
+      .catch(() => res.status(500).jsend.error('Oops, something has gone wrong on the serve'));
```
Comment (joeeasy) [inline server/controllers/notificationController.js]: Great job👍🏻 @emmaadesile, Can you kindly change the spelling `serve` to `server`

File: server/controllers/notificationController.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
```
Comment (joeeasy) [inline server/controllers/notificationController.js]: Same applies here.

**Review by johngorithm — CHANGES_REQUESTED** [review]
Good Job 👍 
Please attend to the pieces of feedback from **houndCI bot**

File: server/controllers/notificationController.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+          receiverId: req.currentUser.id
+        }
+      })
+      .then((notify) => {
+        if (!notify) {
+          return res.redirect('/api/v1/notifications');
+        }
+        notify.isRead = true;
+        notify.save()
+          .catch(() => res.status(500).jsend.error('Oops, something has gone wrong on the serve'));
```
Comment (joeeasy) [inline server/controllers/notificationController.js]: Seems like they're a couple of the same wrong spellings here. Please do look into it.

**Review by joeeasy — CHANGES_REQUESTED** [review]
Nice job, just try to address the issues hound CI is complaining about and the spelling mistakes.

File: server/controllers/notificationController.js (hunk 2 above)
Comment (emmaadesile) [reply server/controllers/notificationController.js]: Okay.

File: server/controllers/notificationController.js (hunk 4 above)
Comment (emmaadesile) [reply server/controllers/notificationController.js]: I will make the necessary changes

File: package.json (hunk 1 above)
Comment (emmaadesile) [reply package.json]: That was changed while testing to check if the notification was working. I'll change it back.

File: server/helpers/notify.js [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        bulkData.push({
+          receiverId: article.userId,
+          actorId: req.currentUser.id,
+          action: `${req.currentUser.username} ${excerpt} ${article.title}`,
+          notifiable: 'article',
+          notifiableId: article.id
+        });
+        notify.push(`notify-${article.userId}`,
+          `channel-${article.userId}`,
+          { message: `${req.currentUser.username} ${excerpt} ${article.title}` });
```
[the remark this answers is not in the document]
Comment (joeeasy) [reply server/helpers/notify.js]: well done, just fix the `expected newline` from hounch CI

**Review by joeeasy — CHANGES_REQUESTED** [review]

**Review by johngorithm — APPROVED** [review]

**Review by joeeasy — APPROVED** [review]

**Pull request merged**